test/spec/crud/unified/crud-let.yml

```diff
@@ -0,0 +1,221 @@
+# NOTE: Not yet committed upstream as a spec test!
+
+description: "crud-let"
+
+schemaVersion: "1.0"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name crud-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+
+initialData: &initialData
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - { _id: 1, x: "foo" }
+
+tests:
+  - description: "Find with let option"
+    runOnRequirements:
+      - minServerVersion: "5.0"
+    operations:
+      - name: find
+        object: *collection0
+        arguments:
+          filter: &query0
+            $expr: { $eq: ["$_id", "$$id"] }
+          let: &let0
+            id: 1
+        expectResult:
+          - { x: "foo" }
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                find: *collection0Name
+                filter: *query0
+                let: *let0
+
+  - description: "Find with let option unsupported (server-side error)"
```

Comment:
Looks like the error string returned changes on versions older than 4.2, and on versions older than 3.6 there's even more different error messages. 
I think it is acceptable we set the minumum version to 4.2 since these are temporary anyway.

Reply:
Sure, sounds good :+1: